examples/java/src/test/java/org/apache/beam/examples/WordCountIT.java

```diff
@@ -68,15 +66,14 @@
   public void testE2EWordCount() throws Exception {
     PipelineOptionsFactory.register(WordCountITOptions.class);
     WordCountITOptions options = TestPipeline.testingPipelineOptions().as(WordCountITOptions.class);
+
+    IOChannelUtils.registerStandardIOFactories(options);
     options.setOutput(
-        Joiner.on("/")
-            .join(
-                new String[] {
-                  options.getTempRoot(),
-                  String.format("WordCountIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date()),
-                  "output",
-                  "results"
-                }));
+            IOChannelUtils.resolve(
+                    options.getTempRoot(),
+                    IOChannelUtils.resolve(
+                            String.format("WordCountIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date()),
+                            IOChannelUtils.resolve("output", "results"))));
```

Comment:
I'm running WordCountIT on linux and OSX. Both of them add a "\" between "output" and "results", which is the same as original code. Do I miss anything? 

But having a method IOChannelUtils.resolve(String base, String ... paths) definitely helps. I'll do it in separate PR.